### PR TITLE
Speed up

### PR DESCRIPTION
**Pull request by mvidner**
- https://trello.com/c/IiJYLiBk/1110-3-speedup-reading-services-state-services-manager 
- https://bugzilla.suse.com/show_bug.cgi?id=1045658

### CODE REVIEW

File: src/modules/services_manager_service.rb [hunk 1]
```diff
@@ -182,7 +180,7 @@ def extract_services
           log.info "  have #{sh}; found #{s.inspect}"
           sh[:enabled] = s && s.enabled?
           log.info "  enabled #{sh[:enabled]}"
-          sh[:active] = is_active?(name)
+          sh[:active] = s && s.active?
```
Comment (jreidinger) [inline src/modules/services_manager_service.rb]: just be sure it does not break in installation, as if I remember correctly we introduce that is-active for some broken (not yet known ) states.

**Comment by mvidner** [comment]
This depends on https://github.com/yast/yast-yast2/pull/613

File: src/modules/services_manager_service.rb (hunk 1 above)
Comment (mvidner) [reply src/modules/services_manager_service.rb]: Yes, the old fix is preserved: https://github.com/yast/yast-yast2/pull/613/files#diff-f44cd7355b2160b663b0bb24f66dbe9aR40

File: src/clients/default_target_proposal.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ def initialize
     def call args
       function = args.shift.to_s
       #TODO implement behaviour if force_reset parameter provided
-      force_reset = !!args.shift
+      _force_reset = !!args.shift
```
Comment (jreidinger) [inline src/clients/default_target_proposal.rb]: you can remove this call completelly, as args is not used.

File: src/clients/services_proposal.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@ def call args
       function = args.shift.to_s
       service_id = args.find {|i| i == 'chosen_id'}.to_s
       #TODO implement behaviour if force_reset parameter provided
-      force_reset = !!(args.find {|i| i == 'force_reset'})
+      _force_reset = !!(args.find {|i| i == 'force_reset'})
```
Comment (jreidinger) [inline src/clients/services_proposal.rb]: same here.

File: src/modules/services_manager_service.rb [hunk 4]
```diff
@@ -12,31 +12,77 @@ class ServicesManagerServiceClass < Module
     LIST_UNIT_FILES_COMMAND = 'systemctl list-unit-files --type service'
     LIST_UNITS_COMMAND      = 'systemctl list-units --all --type service'
     STATUS_COMMAND          = 'systemctl status'
-    IS_ACTIVE_COMMAND       = 'systemctl is-active'
+    # WTF, duplicated in Yast::Systemctl
```
Comment (jreidinger) [inline src/modules/services_manager_service.rb]: maybe use XXX: or FIXME: so it is easier to search.

File: src/modules/services_manager_service.rb [hunk 5]
```diff
@@ -12,31 +12,77 @@ class ServicesManagerServiceClass < Module
...
+      :active         => nil,
+      :loaded         => false,
+      :description    => nil
     }
 
+    # FIXME: this is a mixture of LoadState (unit) and UnitFileState (unit file)
     module Status
+      # LoadState
       LOADED     = 'loaded'
+      # LoadState
```
Comment (jreidinger) [inline src/modules/services_manager_service.rb]: ???

File: src/modules/services_manager_service.rb [hunk 6]
```diff
@@ -12,31 +12,77 @@ class ServicesManagerServiceClass < Module
...
       NOTFOUND   = 'not-found'
+      # masked is both a LoadState and a UnitFileState :-/
       MASKED     = 'masked' # The service has been marked as completely unstartable, automatically or manually.
+      # UnitFileState
       STATIC     = 'static' # The service is missing the [Install] section in its init script, so you cannot enable or disable it.
```
Comment (jreidinger) [inline src/modules/services_manager_service.rb]: is trailing comment shown in yardoc?

File: src/modules/services_manager_service.rb [hunk 7]
```diff
@@ -59,39 +105,35 @@ def read
...
+        out = SCR.Execute(Path.new('.target.bash_output'), command)['stdout']
+        out.lines
       end
 
       def load_unit_files
-        list_unit_files['stdout'].each_line do |line|
+        list_unit_files.each do |line|
           service, status = line.split(/[\s]+/)
           service.chomp! SERVICE_SUFFIX
+          # Unit template, errors out when inquired with `systemctl show`
```
Comment (jreidinger) [inline src/modules/services_manager_service.rb]: any reference where to get more info about it?

File: src/modules/services_manager_service.rb [hunk 8]
```diff
@@ -127,28 +169,35 @@ def extract_services
...
-          if !services[name][:description] || services[name][:description].empty?
-            # Trying to evaluate description via the show command of systemctl
-            s = SystemdService.find(name)
-            services[name][:description] = SystemdService.find(name).description if s
+        service_names.zip(ss).each do |name, s|
+          sh = services[name] # service hash
+          sh[:enabled] = s && s.enabled?
+          sh[:active] = s && s.active?
+          if !sh[:description] || sh[:description].empty?
+            sh[:description] = s.description if s
```
Comment (jreidinger) [inline src/modules/services_manager_service.rb]: so description is nil if service not found? maybe `""` is better?

File: src/modules/services_manager_service.rb (hunk 5 above)
Comment (mvidner) [reply src/modules/services_manager_service.rb]: Yes? What should I fix or explain?

File: src/modules/services_manager_service.rb (hunk 6 above)
Comment (mvidner) [reply src/modules/services_manager_service.rb]: Only if there is no comment on the preceding line. Fixing.

File: src/modules/services_manager_service.rb (hunk 7 above)
Comment (mvidner) [reply src/modules/services_manager_service.rb]: Added

File: src/modules/services_manager_service.rb (hunk 5 above)
Comment (jreidinger) [reply src/modules/services_manager_service.rb]: why `NOTFOUND = 'not=found'` is commented as LoadState? Can you a slightly explain it? like possible LoadState ( with link to it ) enum value?

Comment (mvidner) [reply src/modules/services_manager_service.rb]: OK, the module comment says 
>     # FIXME: this is a mixture of LoadState (unit) and UnitFileState (unit file)

so is it clearer if I change that to
>     # FIXME: this is a mixture of
>     # LoadState (the LOAD column of systemctl list-units) and
>     # UnitFileState (STATE of systemctl list-unit-files)

It's a fixme, I did not want to shave too many yaks.

File: src/modules/services_manager_service.rb (hunk 8 above)
Comment (mvidner) [reply src/modules/services_manager_service.rb]: Uh, I don't see a simple way to do this.

Comment (jreidinger) [reply src/modules/services_manager_service.rb]: something like `sh[:description] = s ? s.description : ""` won't work?

**Pull request merged**